src/pipx/commands/upgrade.py

```diff
@@ -34,21 +32,44 @@ def upgrade(
 
     venv = Venv(venv_dir, verbose=verbose)
 
-    old_version = venv.get_venv_metadata_for_package(package).package_version
+    package_metadata = venv.package_metadata[package]
+    if package_metadata.package_or_url is not None:
+        package_or_url = package_metadata.package_or_url
+        old_version = package_metadata.package_version
+        include_apps = package_metadata.include_apps
+        include_dependencies = package_metadata.include_dependencies
+    else:
+        # fallback if no metadata
+        package_or_url = package
+        old_version = ""
+        include_apps = True
+        include_dependencies = False
+
+    if package == "pipx":
+        package_or_url = PIPX_PACKAGE_NAME
 
     # Upgrade shared libraries (pip, setuptools and wheel)
     venv.upgrade_packaging_libraries(pip_args)
 
-    venv.upgrade_package(package_or_url, pip_args)
-    new_version = venv.get_venv_metadata_for_package(package).package_version
+    venv.upgrade_package(
+        package,
+        package_or_url,
+        pip_args,
+        include_dependencies=include_dependencies,
+        include_apps=include_apps,
+        is_main_package=True,
+    )
+    # TODO 20191026: Should we upgrade injected packages also?
```

Comment:
What is your workflow for TODOs? I prefer TODOs to be associated with an issue.

Reply:
I agree.  My flow is that these are things that I notice along the way that I don't want to forget while I'm still busy coding.  I always intended to get rid of the ones that remain before the final commit and make them actual Issues instead.
What is your preference?  Leave any TODOs in the code?  Or just make sure anything left is put into an Issue?  I'd be happy to do whatever.

Reply:
Updated remaining TODO to have Issue # for tracking.
Done.